raft/server.cc

```diff
@@ -503,6 +503,8 @@ future<> server_impl::add_entry(command command, wait_type type, seastar::abort_
             if (std::holds_alternative<raft::entry_id>(reply)) {
                 co_return co_await wait_for_entry(std::get<raft::entry_id>(reply), type, as);
             }
+            // FIXME: shouldn't we check for commit_status_unknown?
```

Comment:
'execute_add_entry` cannot return `commit_status_unknown` because waiting is done locally. `execute_modify_config` OTOH commits non joint entry internally and wait for it, so it may return `commit_status_unknown`.

Side question: why do we wait internally instead of returning the index of non joint entry and wait for it outside? The answer seams to be somewhere here:  28b579248167b48844a01dd8d7c76d61edceac1f

Reply:
> Side question: why do we wait internally instead of returning the index of non joint entry and wait for it outside? The answer seams to be somewhere here: https://github.com/scylladb/scylla/commit/28b579248167b48844a01dd8d7c76d61edceac1f

That's exactly where the answer is.

> execute_add_entry cannot return commit_status_unknown because waiting is done locally. execute_modify_config OTOH commits non joint entry internally and wait for it, so it may return commit_status_unknown.

Makes sense. In that case I guess an assert would be appropriate.